### PR TITLE
fix: json yaml conflic umarshling template ref to DAG task - RK-19034

### PR DESCRIPTION
**Pull request by gosharo**
Signed-off-by: Gosha <gosha@rookout.com>

### Description

Please provide a brief description of the changes made in this pull request.

### Related Issue(s)

If this pull request addresses or relates to any open issues, please mention them here using the syntax `Fixes #<issue_number>` or `Resolves #<issue_number>`.

### Checklist

Before submitting this pull request, please ensure that you have completed the following tasks:

- [x] Reviewed the [Contributing Guidelines](../docs/CONTRIBUTING.md) for the Piper project.
- [x] Ensured that your changes follow the [coding guidelines and style](../docs/CONTRIBUTING.md#coding-guidelines) of the project.
- [x] Run the tests locally and made sure they pass.
- [x] Updated the relevant documentation, if applicable.

### Testing Instructions

Please provide clear instructions on how to test and verify the changes made in this pull request.

### Additional Information

Add any additional information or context that would be helpful in understanding and reviewing this pull request.

### CODE REVIEW

**Review by nissim-infra — APPROVED** [review]

File: pkg/workflow_handler/workflows_utils_test.go [hunk 1]
```diff
@@ -39,3 +39,75 @@ func TestAddFilesToTemplates(t *testing.T) {
 	expectedScript := "echo \"wellcome to {{ workflow.parameters.global }}\necho \"{{ inputs.parameters.message }}\"\n"
 	assert.Equal(expectedScript, template[0].Script.Source)
 }
+
+func TestCreateDAGTemplate(t *testing.T) {
+	// Test case 1: Empty file list
```
Comment (qadiludmer) [inline pkg/workflow_handler/workflows_utils_test.go:44]: Add the following tests
- handling yaml with wrong fields (not matching to the DAG struct)
- handling yaml with syntax error

**Review by qadiludmer — CHANGES_REQUESTED** [review]
see comments

File: pkg/workflow_handler/workflows_utils_test.go (hunk 1 above)
Comment (gosharo) [reply pkg/workflow_handler/workflows_utils_test.go:44]: done

**Review by nissim-infra — APPROVED** [review]

**Pull request merged**